### PR TITLE
Adding a new "Replace" normalizer that takes a string and replaces it

### LINKED ISSUE #414: Add a normalizer for removing double single quotes
Transformers tokenizer_albert code contains this line:
```python
line = line.replace("``", '"').replace("''", '"')
```

We currently have no way to handle this with normalizers. We need it to be able to emulate this tokenizer.

Either a specialized one, or regexp ?

### PR DESCRIPTION
**Pull request by Narsil**
with another String (for now).
Fixes #414

### CODE REVIEW

File: bindings/python/src/normalizers.rs [hunk 1]
```diff
@@ -310,6 +312,18 @@ impl PyPrecompiled {
     }
 }
 
+#[pyclass(extends=PyNormalizer, module = "tokenizers.normalizers", name=Replace)]
+pub struct PyReplace {}
+#[pymethods]
+impl PyReplace {
+    #[new]
+    fn new(pattern: &PyString, content: &PyString) -> PyResult<(Self, PyNormalizer)> {
```
Comment (n1t0) [inline bindings/python/src/normalizers.rs:320]: No need to convert manually, you can do:
```rust
fn new(pattern: String, content: String) -> PyResult<(Self, PyNormalizer)> {
```

File: tokenizers/src/normalizers/replace.rs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+impl Replace {
+    pub fn new(pattern: String, content: String) -> Self {
+        Self { pattern, content }
+    }
+}
+
+impl Normalizer for Replace {
+    fn normalize(&self, normalized: &mut NormalizedString) -> Result<()> {
+        let pattern: &str = &self.pattern;
+        normalized.replace(pattern, &self.content)?;
```
Comment (n1t0) [inline tokenizers/src/normalizers/replace.rs:22]: You can directly return `normalized.replace(...)` here.

**Review by n1t0 — CHANGES_REQUESTED** [review]
Nice! We can probably already make it generic over `Pattern` in Rust, we just miss some stuff to implement the Python bindings at the moment.

**Comment by Narsil** [comment]
Given that it's relatively hard to put directly a pattern here, can we move this forward for now so we can get `tokenizers` fast into transformers first, then figure out how to make it generic in another issue/PR ?

**Pull request merged**